Assets/HotmaxEditorVR/propertiesMenu/propertiesMenu.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+public class propertiesMenu : MonoBehaviour
+{
+    //-----DEPENDENCIES-----\\
+    //--All materials must be in a Materials folder in the "Resources"
+    //--prefab called "lightSource" must be in the Resources folder
+    //--All imported textures must be read/writable
+
+    //--local refs
+    private GameObject _selectedObject = stateManager.selectedObject;
+    private stateManager.editorModes _editorMode = stateManager.editorMode;
+
+
+    //--add all materials in object to list
+    private List<string> materialNames = new List<string>();
+    private List<List<string>> chunkedMaterialList = new List<List<string>>();
+    private int currentMaterialsPage = 0;
+
+    //--refs to children objects
+    public GameObject materialListHolder;
+    public GameObject slider;
+    public TextMesh physicsText;
+
+    private Rigidbody selectedObjectRb;
+
+    //--slider stuff
+    private bool sliderMoving = false;
+    float sliderUnit;
+
+    private bool massScaleHit = false;
+
+    void Start()
```

Comment:
would onEnable + onDisable makes more sense here for adding/removing watchers?

Reply:
OnEnable and OnDisable in this instance won't update the selected object because it isn't listening while the menu object is non-active.